bikes/views.py

```diff
@@ -82,3 +88,80 @@ def create(self, request, *args, **kwargs):
             {"message": "Bike is not available, it is rented, blocked or reserved"},
             status=status.HTTP_422_UNPROCESSABLE_ENTITY,
         )
+
+
+class ReservationsViewSet(viewsets.ModelViewSet):
+    queryset = Bike.objects.filter(status=BikeStatus.reserved)
+    serializer_class = ReserveBikeSerializer
+
+    def get_serializer_class(self):
+        if self.action == "create":
+            return ReserveIdSerializer
+        else:
+            return ReserveBikeSerializer
+
+    def handle_exception(self, exc):
+        if isinstance(exc, Http404):
+            return Response(
+                {"message": "Bike not reserved"},
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+
+        return super().handle_exception(exc)
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            reserved_bike = Bike.objects.get(id=request.data.get("id"))
+        except Bike.DoesNotExist:
+            return Response(
+                {"message": "Bike not found"}, status=status.HTTP_404_NOT_FOUND
+            )
+        if (
+            reserved_bike.status != BikeStatus.available
+            or reserved_bike.station.state == StationState.blocked
+        ):
+            return Response(
+                {
+                    "message": "Bike already blocked, rented or reserved, or station is blocked"
+                },
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+        # TODO(kboryczka): uncomment once blocking users is introduced
+        # user = somehow_get_user_from_token_in_headers(request)
+        # if user.status == UserStatus.blocked:
+        #     return Response(
+        #         {"message": "User is blocked"},
+        #         status=status.HTTP_403_FORBIDDEN,
+        #     )
+        time = timezone.now()
+        Reservation.objects.create(
+            bike=reserved_bike,
+            reserved_at=time,
+            reserved_till=time + timezone.timedelta(minutes=30),
+        )
+
+        reserved_bike.status = BikeStatus.reserved
+        reserved_bike.save()
+        return Response(
+            data=ReserveBikeSerializer(reserved_bike).data,
+            status=status.HTTP_201_CREATED,
```

Comment:
To pytanie osoby niedoświadczonej, w Response nie powinno być jeszcze headers? Czy django samo to ogarenie?

Reply:
Musiałem usunąć headers, gdyż mi nie przechodził test poprawności stylistycznej, ponieważ zmienna headers nie jest używana w tej funkcji w ogóle.

Reply:
@KrolikJakub Te headers generują się automatycznie, to ja kiedyś niechcący wrzuciłem ustawianie ich explicite. W ramach jakiegoś refaktoru trzeba będzie je pousuwać z innych metod też.